packages/cli/src/deploy/deployWorld.ts

```diff
@@ -24,13 +24,7 @@ export async function deployWorld(
     args: [salt],
   });
 
-  debug("waiting for world deploy");
-  const receipt = await waitForTransactionReceipt(client, { hash: tx });
-  if (receipt.status !== "success") {
-    console.error("world deploy failed", receipt);
-    throw new Error("world deploy failed");
-  }
-
+  const [receipt] = await waitForTransactions({ client, hashes: [tx], debugLabel: "world deploy" });
```

Comment:
iirc I intentionally didn't replace this one with `waitForTransactions` because I wanted to throw a more specific error

Reply:
reason i changed this was bc i changed polling settings in `waitForTransactions` and would have had to duplicate the changes here, but moved that to #3198 so could also move this change there